sesame/src/main/scala/org/w3/banana/sesame/SesameTransactor.scala

```diff
@@ -0,0 +1,28 @@
+package org.w3.banana.sesame
+
+import org.openrdf.repository.RepositoryConnection
+import org.w3.banana.{Transactor, SparqlUpdate, RDFStore}
+
+import scala.util.{Failure, Try}
+
+trait SesameTransactor extends Transactor[Sesame,RepositoryConnection] {
+
+  def r[T](conn: RepositoryConnection, body: => T): Try[T] = Try{
+    val result = body
+    conn.close()
+    result
+  }
+
+  def rw[T](conn: RepositoryConnection, body: => T): Try[T] = Try{
```

Comment:
We have to provide a consistent semantics at the level of a read/write transaction.